src/js/measurements.ts

```diff
@@ -0,0 +1,159 @@
+    this._lastInterval = timestampInSeconds() * 1000;
+
+    this._iteration();
+  }
+
+  /**
+   * Stops the stall tracking interval, and returns the measurements
+   */
+  public finish(): {
+    stall_count: { value: number };
+    total_stall_time: { value: number };
+    longest_stall: { value: number };
+  } {
+    if (typeof this._timeout === "number") {
+      clearTimeout(this._timeout);
+    }
+
+    return {
+      stall_count: { value: this._stallCount },
+      total_stall_time: { value: this._totalStallTime },
+      longest_stall: { value: this._longestStall },
+    };
+  }
+
+  /**
+   * Iteration of the stall tracking interval. Measures how long the timer strayed from its expected time of running, and how
+   * long the stall is for.
+   */
+  private _iteration(): void {
+    const now = timestampInSeconds() * 1000;
+    const busyTime = now - this._lastInterval;
+
+    if (busyTime >= this._acceptableBusyTime) {
+      const stallTime = busyTime - this._acceptableBusyTime;
+      this._stallCount += 1;
+      this._totalStallTime += stallTime;
+      this._longestStall = Math.max(this._longestStall, stallTime);
+    }
+
+    this._lastInterval = now;
+    this._timeout = setTimeout(
+      this._iteration.bind(this),
+      this._acceptableBusyTime / 5
+    );
+  }
+}
+
+/**
+ * Adds React Native's extensions. Needs to be called after @sentry/tracing's extension methods are added
+ */
+export function _addTracingExtensions(): void {
+  const carrier = getMainCarrier();
+  if (carrier.__SENTRY__) {
+    carrier.__SENTRY__.extensions = carrier.__SENTRY__.extensions || {};
+    if (carrier.__SENTRY__.extensions.startTransaction) {
+      const originalStartTransaction = carrier.__SENTRY__.extensions
+        .startTransaction as StartTransactionFunction;
+
+      /*
+        Overwrites the transaction start and finish to start and finish stall tracking.
+        Preferably instead of overwriting add a callback method for this in the Transaction itself.
+      */
+      const _startTransaction = _patchStartTransaction(
+        originalStartTransaction
+      );
+
+      carrier.__SENTRY__.extensions.startTransaction = _startTransaction;
+    }
+  }
+}
+
+type StartTransactionFunction = (
+  this: Hub,
+  transactionContext: TransactionContext,
+  customSamplingContext?: CustomSamplingContext
+) => Transaction;
+
+/**
+ * Overwrite the startTransaction extension method to start and end stall tracking.
+ */
+const _patchStartTransaction = (
+  originalStartTransaction: StartTransactionFunction
+): StartTransactionFunction => {
+  /**
+   * Method to overwrite with
+   */
+  function _startTransaction(
+    this: Hub,
+    transactionContext: TransactionContext,
+    customSamplingContext?: CustomSamplingContext
+  ): Transaction {
+    const transaction: Transaction = originalStartTransaction.apply(this, [
+      transactionContext,
+      customSamplingContext,
+    ]);
+
+    const stalls = new Stalls();
+    stalls.start();
```

Comment:
If we get multiple concurrent transactions each one will have its own tracker, right>
So multiple jobs being queued for execution to measure latency?

Reply:
Hmm yes, I realized that can have an effect on performance. If we decide to have one global one, it should not be always tracking and only when some transaction is running, and is closed when all the transactions are finished. Should I go with that?

Reply:
It would be awesome to have this. If it doesn't degrade performance in any obvious way it sounds good. It must have an opt-out though so folks can turn this off if it becomes an issue

Reply:
Added this feature.